components/concurrency_manager/src/lib.rs

```diff
@@ -107,11 +107,26 @@ impl ConcurrencyManager {
     ) -> Result<(), E> {
         self.lock_table.check_range(start_key, end_key, check_fn)
     }
+
+    /// Find the minimum start_ts among all locks in memory.
+    pub fn global_min_lock_ts(&self) -> Option<TimeStamp> {
+        let mut min_lock_ts = None;
+        // TODO: The iteration looks not so efficient. It's better to be optimized.
```

Comment:
I think you can use `self.lock_table.iter().min_by_key(...)`.

Reply:
It's a  good idea to implement the iterator. But the efficiency problem I'm afraid about is that it needs to lock a mutex for each item it iterates.